src/fmeval/transforms/transform_pipeline.py

```diff
@@ -0,0 +1,67 @@
+import ray.data
+from typing import List, Union
+
+from fmeval.constants import TRANSFORM_PIPELINE_MAX_SIZE
+from fmeval.exceptions import EvalAlgorithmClientError
+from fmeval.transforms.transform import Transform
+from fmeval.util import require
+
+NestedTransform = Union[Transform, "TransformPipeline"]
+
+
+class TransformPipeline:
+    def __init__(self, nested_transforms: List[NestedTransform]):
+        require(
+            isinstance(nested_transforms, List),
+            "TransformPipeline initializer accepts a list containing Transforms or TransformPipelines.",
+        )
+        seen_keys = set()
+        duplicate_keys = []
+        self.transforms: List[Transform] = []
+        for nested_transform in nested_transforms:
+            if isinstance(nested_transform, Transform):
+                for key in nested_transform.output_keys:
+                    if key in seen_keys:
+                        duplicate_keys.append(key)
+                    else:
+                        seen_keys.add(key)
+                self.transforms.append(nested_transform)
+            elif isinstance(nested_transform, TransformPipeline):
+                self.transforms += nested_transform.transforms
```

Comment:
Question: Would there ever be a case where the user might alter the pipeline after it's initialized, like add, remove, or swap a Transform? If so, this might cause unexpected results if a user edits a nested TransformPipeline and expects the parent TransformPipeline to reflect those changes when they run `execute`, but that doesn't happen since we copy the nested pipeline's Transforms at initialization.

Reply:
Good point. Based on how I'm planning on implementing all of our eval algorithms, I use `TransformPipeline`s such that they never get mutated after initialization. However, you bring up a valid potential use case that I should consider. Thanks for catching this!

Reply:
I've added information about mutating child pipelines to the `TransformPipeline` docstring (in fact, I didn't even write one earlier). Based on how I envision using `TransformPipelines`, I recommend that `TransformPipeline`s not be mutated after initialization, for the reason you mentioned above.